### PR TITLE
Update to Bevy 0.8

### PR DESCRIPTION
**Pull request by zicklag**
- Make sure that we only depend on Bevy features we need
- Temporarily disable Bevy Egui World inspector until it updates to
Bevy 0.8.
- Add Serde default to LayerData so that the RON demo deserializes
correctly without needing to specify default options.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,25 @@ keywords = ["bevy", "gamedev", "parallax", "scrolling", "background"]
...
+    "bevy_core_pipeline",
+    "bevy_sprite",
+    "bevy_asset",
+    "bevy_winit",
+]
+
 [dev-dependencies]
-bevy-inspector-egui = "0.10.0"
+# Temporarily removed until it updates to Bevy 0.8
+# bevy-inspector-egui = "0.11.0"
```
Comment (odecay) [inline Cargo.toml:30]: waiting on https://github.com/jakobhellermann/bevy-inspector-egui/pull/76

**Review by odecay — COMMENTED** [review]
Looks good

File: Cargo.toml (hunk 1 above)
Comment (odecay) [inline Cargo.toml:30]: `bevy-inspector-egui = "0.12.0"`
I think its ready to go

Comment (zicklag) [reply Cargo.toml:30]: Sweet. I'll get to updating this to use it tomorrow.

**Review by cdsupina — APPROVED** [review]
I'm on a road trip for the next week and a half, I won't have time to thoroughly check this. I took a quick look and it looks great to me. Feel free to merge when ready.

**Pull request merged**